src/Npgsql/NpgsqlSchema.cs

```diff
@@ -172,14 +173,17 @@ static DataTable GetDatabases(NpgsqlConnection conn, string?[]? restrictions)
 
             getDatabases.Append("SELECT d.datname AS database_name, u.usename AS owner, pg_catalog.pg_encoding_to_char(d.encoding) AS encoding FROM pg_catalog.pg_database d LEFT JOIN pg_catalog.pg_user u ON d.datdba = u.usesysid");
 
-            using (var command = BuildCommand(conn, getDatabases, restrictions, "datname"))
-            using (var adapter = new NpgsqlDataAdapter(command))
+            using var command = BuildCommand(conn, getDatabases, restrictions, "datname");
+            using var adapter = new NpgsqlDataAdapter(command);
+            if (async)
```

Comment:
You could introduce an internal Fill method accepting an async flag to make all this nicer.